File: src/main/java/org/alxtek/locaiai/entities/Annonce.java
package org.alxtek.locaiai.entities;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.*;
import org.alxtek.locaiai.enums.Disponibilite;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Entity
@AllArgsConstructor @NoArgsConstructor @Builder @Getter @Setter
public class Annonce {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String titre;
    private String description;
    private Double prix;
    @Enumerated(EnumType.STRING)
    private Disponibilite disponibilite;
    private double promotion;
    private LocalDate debutAnnonce;
    private LocalDate finAnnonce;
    @ManyToOne
    @JoinColumn(name = "hote_id")
    @JsonManagedReference
    private Utilisateur hote;
    @JsonManagedReference
    @OneToMany(mappedBy = "annonce", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<Reservation> reservations = new ArrayList<>();
    @ManyToOne
    @JoinColumn(name = "logement_id")
    @JsonBackReference
    private Logement logement;
    @JsonManagedReference
    @OneToMany(mappedBy = "annonce", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<Evaluation> evaluations = new ArrayList<>();
}

File: front-end/src/app/interfaces/annonce.ts
import {Disponibilite} from "../models/disponibilite.enum";
import {User} from "./utilisateur";
import {Logement} from "./logement";

export interface Annonce {
  id: number;
  titre: string;
  description: string;
  prix: number;
  note: number;
  type: string;
  promotion: number;
  disponibilite: Disponibilite;
  debutAnnonce: Date;
  finAnnonce: Date;
  hote: User;
  logement: Logement;
  evaluations: any[];
}
